src/types.h

```diff
@@ -97,6 +98,9 @@ class SizedType
 
 private:
   bool is_signed_ = false;
+  std::shared_ptr<SizedType> element_type_; // for "container" and pointer
```

Comment:
Don't think this needs to be a shared_ptr b/c I don't see SizedType sharing ownership of this. I think we can just store it directly.

Reply:
I was looking for a good way to handle container types. In this case the copy constructor still works but once we introduce a pointer type it would have to recursively copy the whole chain. I was looking at ensuring that types are immutable and that we only "create" a type once (calling `CreateInt(8)` twice should return the same object).

Anyway all that isn't really relevant here so I'll take it out :)